Assets/MixedRealityToolkit/_Core/Utilities/Lines/DataProviders/BaseMixedRealityLineDataProvider.cs

```diff
@@ -0,0 +1,485 @@
+    /// Base class that provides data about a line.
+    /// </summary>
+    /// <remarks>Data to be consumed by other classes like the <see cref="Renderers.BaseMixedRealityLineRenderer"/></remarks>
+    [DisallowMultipleComponent]
+    public abstract class BaseMixedRealityLineDataProvider : MonoBehaviour
+    {
+        private const float MinRotationMagnitude = 0.0001f;
+
+        public float UnClampedWorldLength => GetUnClampedWorldLengthInternal();
+
+        [Range(0f, 1f)]
+        [SerializeField]
+        [Tooltip("Clamps the line's normalized start point. This setting will affect line renderers.")]
+        private float lineStartClamp = 0f;
+
+        /// <summary>
+        /// Clamps the line's normalized start point. This setting will affect line renderers.
+        /// </summary>
+        public float LineStartClamp
+        {
+            get { return lineStartClamp; }
+            set { lineStartClamp = Mathf.Clamp01(value); }
+        }
+
+        [Range(0f, 1f)]
+        [SerializeField]
+        [Tooltip("Clamps the line's normalized end point. This setting will affect line renderers.")]
+        private float lineEndClamp = 1f;
+
+        /// <summary>
+        /// Clamps the line's normalized end point. This setting will affect line renderers.
+        /// </summary>
+        public float LineEndClamp
+        {
+            get { return lineEndClamp; }
+            set { lineEndClamp = Mathf.Clamp01(value); }
+        }
+
+        [SerializeField]
+        [Tooltip("Transform to use when translating points from local to world space. If null, this object's transform is used.")]
+        private Transform customLineTransform;
+
+        /// <summary>
+        /// Transform to use when translating points from local to world space. If null, this object's transform is used.
+        /// </summary>
+        public Transform LineTransform
+        {
+            get { return customLineTransform != null ? customLineTransform : transform; }
+            set { customLineTransform = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("Controls whether this line loops \nNote: some classes override this setting")]
+        private bool loops = false;
+
+        /// <summary>
+        /// Controls whether this line loops
+        /// </summary>
+        /// <remarks>Some classes override this setting.</remarks>
+        public virtual bool Loops
+        {
+            get { return loops; }
+            set { loops = value; }
+        }
+
+        [SerializeField]
+        [Tooltip("The rotation mode used in the GetRotation function. You can visualize rotations by checking Draw Rotations under Editor Settings.")]
+        private LineRotationType rotationType = LineRotationType.Velocity;
+
+        /// <summary>
+        /// The rotation mode used in the GetRotation function. You can visualize rotations by checking Draw Rotations under Editor Settings.
+        /// </summary>
+        public LineRotationType RotationType
```

Comment:
Does this need to be a specific Line rotation type?  Can it not be genericised for just RotationType?

Reply:
Pretty specific, as it only pertains to the line rotation. (Unless that's what you're requesting is to rename it.)